tests/drivers/uart/uart_async_api/src/test_uart_async.c

```diff
@@ -67,26 +73,44 @@ void test_single_read_setup(void)
 
 void test_single_read(void)
 {
+#if CONFIG_NOCACHE_MEMORY
+	const uint8_t TX_BUF[] = "test";
+
+	static __aligned(16) uint8_t tx_buf[16] __used
+		__attribute__((__section__(".nocache")));
+	static __aligned(16) uint8_t rx_buf[16] __used
+		__attribute__((__section__(".nocache.dma")));
+
```

Comment:
I wonder why this variable need to be aligned to 16 bytes. For an uncached section, there is no need for specific alignment. I also wonder why the difference in section `".nocache"` vs `".nocache.dma"'. Why not simply using the `__nocache` macro?

Reply:
I tried building the test with a special "USER NOCACHE MEMORY" area where user buffers are defined for the DMA transfers.  The CONFIG_NOCACHE_MEMORY could be ANDed with CONFIG_USERSPACE  to map this area.

Reply:
We can see 4 different approaches, where item 1 and 2 are feasible
1. allocate a user space NOCACHE MEMORY 
2. Invalidate/clean Cache memory in _each_ driver with DMA (adapt for each peripheral driver with dma)
3. Invalidate/clean Cache memory in the dma drivers : once for all the pepriphera
4. allocate a user space NOCACHE MEMORY for the dma driver and memcpy in each client buffer

Reply:
should data be aligned on data aligned on `DCACHE_LINE_SIZE` or on `arch_dcache_line_size_get() `

Reply:
In the NOCACHE memory case, I don't think you need any alignment.

Reply:
> We can see 4 different approaches, where item 1 and 2 are feasible
> 
>     1. allocate a user space NOCACHE MEMORY
> 
>     2. Invalidate/clean Cache memory in _each_ driver with DMA (adapt for each peripheral driver with dma, like in #27911)
> 
>     3. Invalidate/clean Cache memory in the dma drivers : once for all the peripheral

If going the cache management operation way, the option 3 would be great to avoid redoing the job for each driver.
 
>     4. allocate a user space NOCACHE MEMORY for the dma driver and memcpy in each client buffer

This option also works with DTCM memory or even with standard memory with cache management operation. Note that this could be made optional if the driver doesn't communicate directly with the user space, but rather with a subsystem and the subsystem buffer can be used directly (example network packets for an Ethernet driver).